test/asynchronous/test_async_cancellation.py

```diff
@@ -0,0 +1,136 @@
+# Copyright 2025-present MongoDB, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Test that async cancellation performed by users clean up resources correctly."""
+from __future__ import annotations
+
+import asyncio
+import sys
+from test.utils import async_get_pool, delay, one
+
+sys.path[0:0] = [""]
+
+from test.asynchronous import AsyncIntegrationTest, async_client_context, connected
+
+
+class TestAsyncCancellation(AsyncIntegrationTest):
+    async def test_async_cancellation_closes_connection(self):
+        client = await self.async_rs_or_single_client(maxPoolSize=1)
```

Comment:
I think all of these tests can reuse the global test client instead of creating new ones.